config/cohere.yaml

```diff
@@ -0,0 +1,73 @@
+# ==================================================================
+#            Configuration file for Canopy Server with Cohere.
+# ==================================================================
+
+# ---------------------------------------------------------------------------------
+system_prompt: &system_prompt |
+  Use the documents to answer the user question at the next messages. The documents are retrieved from a knowledge
```

Comment:
+1 here. When specifying `documents` in the SDK, our RAG preamble gets used here to generate an appropriate response with the given documents (somewhat similar to how `query_builder_prompt` is not used in `cohere.yaml` due to our `search_queries` boolean). In Cohere case, `system_prompt` should be an optional knob for someone that wants to experiment with a custom RAG preamble.

The only issue here I believe is that there is a `DEFAULT_SYSTEM_PROMPT` in `chat_engine.py` that gets used if this is omitted? Can work something out here.

Reply:
Canopy allows users to change the system prompt, even in the RAG case, so they can add their own tweaks like avoiding specific topics, using specific tones etc.  
Unless there is a very strong incentive to do otherwise, we would like to keep this behavior.

@aalpat4 \ @giannis2two can we simply put the default Cohere RAG prompt here in the config and leave it transparent for users to change?

Your suggestion is a bit problematic, as it breaks Canopy's design principles. Note that this `system_prompt` key configures the `ChatEngine` itself, not the underlying `LLM` class.  
The `CohereLLM` class is considered a stand alone, **stateless** LLM wrapper, which takes   `system_prompt` (==preamble) and `context` (==`documents`) inputs **per `chat()` call**. Theoretically the same `LLM` instance might be used to do RAG in a one `chat()` call, and a completely different task in the next call (with a different prompt, without documents etc).

Reply:
Understood thanks for the details here @igiloh-pinecone. Let us check internally regarding `can we simply put Cohere's RAG prompt here`. Can think of another potential solution here as well. Will come back here.